contributors/devel/sig-instrumentation/logging.md

```diff
@@ -1,37 +1,73 @@
-## Logging Conventions
+## Logging
 
-The following conventions for the klog levels to use.
-[klog](http://godoc.org/github.com/kubernetes/klog) is globally preferred to
-[log](http://golang.org/pkg/log/) for better runtime control.
+This document will provide an overview on how Kubernetes community recommendation on logging. 
+Following recommendation should be followed in https://github.com/kubernetes/kubernetes, however any Kubernetes or external project is fill to follow guidelines below as most of the ideas can be applied more broadly.
 
-Shared libraries, such as `client-go`, should not use `klog.ErrorS()` and `klog.InfoS()`,
+### Logging in Kubernetes
+
+Kubernetes project uses [klog](https://github.com/kubernetes/klog) for logging.
+This library was created as a permanent fork of glog, as the original project was no longer developed. 
+However, should not be used by external projects as it is mostly developed to maintain backward compatibility.
+With introduction of [Structured Logging](https://github.com/kubernetes/enhancements/issues/1602) Kubernetes is being migrated to [logr](https://github.com/go-logr/logr) interface, any changes in klog are done to ensure smooth transition.
+
+For projects that want to integrate with Kubernetes or write same log format we recommend using [klogr](https://github.com/kubernetes/klog/tree/main/klogr), reimplementation of klog format that exposes [logr](https://github.com/go-logr/logr) interface.
```

Comment:
Is this correct?
```suggestion
For projects that want to integrate with Kubernetes or write same log format we recommend using [klogr](https://github.com/kubernetes/klog/tree/main/klogr). The `klogr` library exposes an interface that
follows [logr](https://github.com/go-logr/logr) conventions, and emits log entries that are equivalent to `klog`.
```

Reply:
I don't think we should recommend using klogr, or at least not without further explanations. For example, a library should accept a `logr.Logger` instance and let the caller decide whether that is klogr or something else.

But right now we are not ready yet to propose *how* such a library should accept such a logger - we need a decision about https://github.com/kubernetes/enhancements/pull/3078 first.